include/nbl/builtin/hlsl/tgmath.hlsl

```diff
@@ -39,6 +39,38 @@ NBL_CONSTEXPR_INLINE_FUNC bool isInf(T val)
 	return (tmp & (~ieee754::traits<AsFloat>::signMask)) == ieee754::traits<AsFloat>::inf;
 }
 
+#ifdef __HLSL_VERSION
+#define INTRINSIC_FUNC_NAMESPACE spirv
+#else
+#define INTRINSIC_FUNC_NAMESPACE std
+#endif
+
+#define DEFINE_IS_NAN_SPECIALIZATION(TYPE)\
+template<>\
+inline bool isNaN<TYPE>(TYPE val)\
+{\
+	INTRINSIC_FUNC_NAMESPACE::isnan(val);\
+}\
+
+#define DEFINE_IS_INF_SPECIALIZATION(TYPE)\
+template<>\
+inline bool isInf<TYPE>(TYPE val)\
+{\
+	INTRINSIC_FUNC_NAMESPACE::isinf(val);\
+}\
```

Comment:
fun fact, GCC and Clang break both `std::isinf` and `std::isnan` with `-ffast-math` so you always need to use the above bitpattern comparison based `impl` functions on the C++ path

Reply:
Imma mark as resolved but needs to be addressed in #803 or #804